src/Modal/ModalLayer.jsx

```diff
@@ -52,29 +52,31 @@ function ModalLayer({
     };
   }, [isOpen]);
 
-  if (!isOpen) {
+  if (!isOpen && !isHidden) {
     return null;
   }
 
   const onClickOutside = !isBlocking ? onClose : null;
 
   return (
-    <ModalContextProvider onClose={onClose} isOpen={isOpen} isBlocking={isBlocking}>
+    <ModalContextProvider onClose={onClose} isOpen={isOpen} isHidden={isHidden} isBlocking={isBlocking}>
       <Portal>
         <FocusOn
           allowPinchZoom
           scrollLock
-          enabled={isOpen}
+          enabled={isOpen && !isHidden}
           onEscapeKey={onClose}
           onClickOutside={onClickOutside}
-          className={classNames(
-            'pgn__modal-layer',
-            zIndex ? `zindex-${zIndex}` : '',
-          )}
+          className={classNames({
+            'pgn__modal-layer': isOpen && !isHidden,
```

Comment:
[inform] Conditionally include the `pgn__modal-layer` class name only when the modal contents and the backdrop should actually be visible. Without making this class name conditional, the resulting `div` from `FocusOn` acts a non-clickable overlay.